src/bindings/fluid_ladspa.c

```diff
@@ -265,6 +260,55 @@ void delete_fluid_ladspa_fx(fluid_ladspa_fx_t *fx)
     FLUID_FREE(fx);
 }
 
+/**
+ * Add host buffers to the LADSPA engine.
+ *
+ * @note The size of the buffers pointed to by the buffers array must be
+ * at least as large as the buffer size given to new_fluid_ladspa_fx.
+ *
+ * @param fx LADSPA fx instance
+ * @param prefix common name prefix for the created nodes
+ * @param num_buffers number of of buffers buffer array
+ * @param buffers array of pointers to buffers
+ * @return FLUID_OK on success, otherwise FLUID_FAILED
+ */
+int fluid_ladspa_add_host_ports(fluid_ladspa_fx_t *fx, const char *prefix,
+        int buffer_count, fluid_real_t *buffers[])
+{
+    int i;
+    char name[99];
+
+    LADSPA_API_ENTER(fx);
+
+    if (fluid_ladspa_is_active(fx))
+    {
+        LADSPA_API_RETURN(fx, FLUID_FAILED);
+    }
+
+    /* Create nodes for all channels */
+    for (i = 0; i < buffer_count; i++)
+    {
+        /* If there is more than one buffer, then append a 1-based index to each node name */
+        if (buffer_count > 1) {
+            FLUID_SNPRINTF(name, sizeof(name), "%s%d", prefix, (i + 1));
+        }
+        else
+        {
+            FLUID_STRNCPY(name, prefix, sizeof(name));
+        }
+
+        if (new_fluid_ladspa_node(fx, name,
+                    FLUID_LADSPA_NODE_AUDIO | FLUID_LADSPA_NODE_HOST,
+                    buffers[i]) == NULL)
+        {
+            return FLUID_FAILED;
+        }
+    }
+
+    return FLUID_OK;
```

Comment:
Direct return without `LADSPA_API_RETURN()` intended?

Reply:
Definitely not, well spotted. Seems like I haven't tested simultaneous access to the API from multiple threads, otherwise this should have tripped me up.